```
/__w/spectre/spectre/src/Domain/CoordinateMaps/TimeDependent/Rotation.hpp:105:8: error: method 'is_identity' can be made static [readability-convert-member-functions-to-static,-warnings-as-errors]
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
  bool is_identity() const noexcept { return false; }
2 warnings treated as errors
       ^             ~~~~~~
  static
```
I thought there was a discussion of this somewhere recently.  Is it being handled elsewhere or was there a decision on what to do or anything?

--- a/src/Domain/DomainHelpers.hpp
+++ b/src/Domain/DomainHelpers.hpp
@@ -383,7 +383,7 @@ template <size_t VolumeDim>
 FaceCornerIterator<VolumeDim>::FaceCornerIterator(
     Direction<VolumeDim> direction) noexcept
     : direction_(std::move(direction)),
-      index_(direction.side() == Side::Upper
+      index_(direction_.side() == Side::Upper
                  ? two_to_the(direction_.dimension())
                  : 0) {
   for (size_t i = 0; i < VolumeDim; ++i) {

--- a/tests/Unit/Domain/Creators/Test_AlignedLattice.cpp
+++ b/tests/Unit/Domain/Creators/Test_AlignedLattice.cpp
@@ -132,6 +132,8 @@ SPECTRE_TEST_CASE("Unit.Domain.Creators.AlignedLattice", "[Domain][Unit]") {
                         {{72.0, 91.0}, {{6, 7}}}, {{70.0, 92.0}, {{2, 3}}},
                         {{71.0, 92.0}, {{2, 3}}}, {{72.0, 92.0}, {{6, 7}}}};
     const auto domain = refined_domain->create_domain();
+    test_initial_domain(domain, refined_domain->initial_refinement_levels());
+
     const auto& blocks = domain.blocks();
     const auto extents = refined_domain->initial_extents();
     REQUIRE(blocks.size() == extents.size());
@@ -153,8 +155,8 @@ SPECTRE_TEST_CASE("Unit.Domain.Creators.AlignedLattice", "[Domain][Unit]") {
 
   {
     // Expected domain refinement:
-    // 25 25 36
-    // 25 35 36
+    // 25 25 46
+    // 25 35 46
     // 25 XX 35
     const auto refined_domain =
         TestHelpers::test_factory_creation<DomainCreator<2>>(
@@ -169,17 +171,19 @@ SPECTRE_TEST_CASE("Unit.Domain.Creators.AlignedLattice", "[Domain][Unit]") {
             "    Refinement: [3, 5]\n"
             "  - LowerCornerIndex: [2, 1]\n"
             "    UpperCornerIndex: [3, 3]\n"
-            "    Refinement: [3, 6]");
+            "    Refinement: [4, 6]");
     std::unordered_set<
         std::pair<std::vector<double>, std::array<size_t, 2>>,
         boost::hash<std::pair<std::vector<double>, std::array<size_t, 2>>>>
         expected_blocks{{{70.0, 90.0}, {{2, 5}}}, {{72.0, 90.0}, {{3, 5}}},
                         {{70.0, 91.0}, {{2, 5}}}, {{71.0, 91.0}, {{3, 5}}},
-                        {{72.0, 91.0}, {{3, 6}}}, {{70.0, 92.0}, {{2, 5}}},
-                        {{71.0, 92.0}, {{2, 5}}}, {{72.0, 92.0}, {{3, 6}}}};
+                        {{72.0, 91.0}, {{4, 6}}}, {{70.0, 92.0}, {{2, 5}}},
+                        {{71.0, 92.0}, {{2, 5}}}, {{72.0, 92.0}, {{4, 6}}}};
     const auto domain = refined_domain->create_domain();
-    const auto& blocks = domain.blocks();
     const auto refinement_levels = refined_domain->initial_refinement_levels();
+    test_initial_domain(domain, refinement_levels);
+
+    const auto& blocks = domain.blocks();
     REQUIRE(blocks.size() == refinement_levels.size());
     for (size_t i = 0; i < blocks.size(); ++i) {
       const auto location =

--- a/tests/Unit/Helpers/Domain/DomainTestHelpers.cpp
+++ b/tests/Unit/Helpers/Domain/DomainTestHelpers.cpp
@@ -143,17 +143,14 @@ void test_domain_connectivity(
 template <size_t AllowedDifference>
 SPECTRE_ALWAYS_INLINE void check_if_levels_are_within(
     const size_t level_1, const size_t level_2) noexcept {
+  // clang-tidy complains about a catch-internal do {} while (false) loop.
+  // NOLINTNEXTLINE(bugprone-infinite-loop)
   CHECK(level_1 <= level_2 + AllowedDifference);
+  // NOLINTNEXTLINE(bugprone-infinite-loop)
   CHECK(level_2 <= level_1 + AllowedDifference);
 }
 
-template <>
-SPECTRE_ALWAYS_INLINE void check_if_levels_are_within<0>(
-    const size_t level_1, const size_t level_2) noexcept {
-  CHECK(level_1 == level_2);
-}
-
-template <size_t AllowedDifference, size_t VolumeDim>
+template <size_t AllowedTangentialDifference, size_t VolumeDim>
 void test_refinement_levels_of_neighbors(
     const std::unordered_map<ElementId<VolumeDim>, Element<VolumeDim>>&
         elements) noexcept {
@@ -164,6 +161,11 @@ void test_refinement_levels_of_neighbors(
       const auto& neighbors = direction_neighbors.second;
       const auto& orientation = neighbors.orientation();
       for (size_t d = 0; d < VolumeDim; ++d) {
+        // No restriction on perpendicular refinement.
+        if (d == direction_neighbors.first.dimension()) {
+          continue;
+        }
+
         const size_t my_dim_in_neighbor = orientation(d);
         const size_t my_level =
             gsl::at(element_id.segment_ids(), d).refinement_level();
@@ -171,8 +173,8 @@ void test_refinement_levels_of_neighbors(
           const size_t neighbor_level =
               gsl::at(neighbor_id.segment_ids(), my_dim_in_neighbor)
                   .refinement_level();
-          check_if_levels_are_within<AllowedDifference>(my_level,
-                                                        neighbor_level);
+          check_if_levels_are_within<AllowedTangentialDifference>(
+              my_level, neighbor_level);
         }
       }
     }
@@ -451,7 +453,7 @@ void test_initial_domain(const Domain<VolumeDim>& domain,
                          initial_refinement_levels));
   }
   domain::test_domain_connectivity(domain, elements);
-  domain::test_refinement_levels_of_neighbors<0>(elements);
+  domain::test_refinement_levels_of_neighbors<1>(elements);
 }
 
 template <typename DataType, size_t SpatialDim>

--- a/tests/Unit/Helpers/Domain/DomainTestHelpers.hpp
+++ b/tests/Unit/Helpers/Domain/DomainTestHelpers.hpp
@@ -72,8 +72,8 @@ boost::rational<size_t> fraction_of_block_volume(
     const ElementId<VolumeDim>& element_id) noexcept;
 
 // Test that the Elements of the initial domain are connected and cover the
-// computational domain, as well as that neighboring Elements  are at the same
-// refinement level.
+// computational domain, as well as that neighboring Elements' refinement
+// levels do not differ too much.
 template <size_t VolumeDim>
 void test_initial_domain(const Domain<VolumeDim>& domain,
                          const std::vector<std::array<size_t, VolumeDim>>&

